api/src/main/java/net/countercraft/movecraft/lifecycle/Worker.java

```diff
@@ -0,0 +1,13 @@
+package net.countercraft.movecraft.lifecycle;
+
+public interface Worker {
+    default int getDelay(){
+        return 1;
+    }
+
+    boolean isAsync();
+
+    int getPeriod();
```

Comment:
Should we default these to false and 1 second?

Reply:
Looking at the usages internally, they mostly are 1 tick services.  However, for users of the API I think it makes more sense to maintain a 1 second default so as to not provide a possible foot-gun.

Reply:
This isn't really intended for external users/addons - if we do see value here I would probably move this stuff to its own package and build it out a bit more.

Reply:
There's a lot of items in the addons that could utilize this type of thing.  I'd like to see us build it in such a way it can be used internally and externally.